kt/godot-library/src/main/kotlin/godot/registration/Registration.kt

```diff
@@ -284,23 +270,28 @@ class ClassBuilderDsl<T : KtObject>(
         func: KFunction1<T, R>,
         variantType: VariantType,
         returnType: KtFunctionArgument,
-        rpcModeId: Int = 0
+        rpcModeId: Int = 0,
+        rpcCallLocal: Boolean,
+        rpcTransferModeId: Int,
+        rpcChannel: Int,
     ) {
         appendFunction(
             KtFunction0(
                 KtFunctionInfo(
-                    func.name.camelToSnakeCase(),
-                    listOf(),
-                    KtPropertyInfo(
-                        returnType.type,
-                        "",
-                        returnType.className,
-                        PropertyHint.NONE,
-                        "",
-                        true, // always true. Only used for properties
-                        0 // always RPCMode.DISABLED. Only used for properties
+                    name = func.name.camelToSnakeCase(),
+                    _arguments = listOf(),
+                    returnVal = KtPropertyInfo(
+                        _type = returnType.type,
+                        name = "",
+                        className = returnType.className,
+                        _hint = PropertyHint.NONE,
+                        hintString = "",
+                        visibleInEditor = true, // always true. Only used for properties
                     ),
-                    rpcModeId
+                    rpcModeId = rpcModeId,
```

Comment:
I would expect to have a class matching the cpp struct

Reply:
Indeed. I I'mplemented this now by introducing a KtRpcConfig class. This also greatly simplifies the cpp side by using a JavaInstanceWrapper like we do for the Function/PropertyInfo classes